chainerrl/envs/abc.py

```diff
@@ -62,7 +62,7 @@ def reset(self):
         return self.observe()
 
     def step(self, action):
-        if isinstance(action, np.ndarray):
+        if isinstance(action, np.ndarray) and action.dtype.kind == 'f':
```

Comment:
I intended to exclude here the cases like `action = array(3, dtype=int32)`.  But now I think it's agent's responsibility to return a truly scalar int.  I'll rewrite in another way to clarify that the type of the action is wrong.